File: src/main/java/com/github/sorusclient/client/hud/AttachType.java
package com.github.sorusclient.client.hud;

import com.github.sorusclient.client.util.Axis;

public class AttachType {

    private final double selfSide, otherSide;
    private final Axis axis;

    private AttachType() {
        this(0, 0, Axis.X);
    }

    public AttachType(double selfSide, double otherSide, Axis axis) {
        this.selfSide = selfSide;
        this.otherSide = otherSide;
        this.axis = axis;
    }

    public double getSelfSide() {
        return selfSide;
    }

    public double getOtherSide() {
        return otherSide;
    }

    public Axis getAxis() {
        return axis;
    }

    public AttachType reverse() {
        return new AttachType(this.otherSide, this.selfSide, this.axis);
    }

}

File: src/main/kotlin/com/github/sorusclient/client/hud/AttachType.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.hud

import com.github.sorusclient.client.util.Axis

class AttachType(val selfSide: Double, val otherSide: Double, val axis: Axis) {

    private constructor() : this(0.0, 0.0, Axis.X)

    fun reverse(): AttachType {
        return AttachType(otherSide, selfSide, axis)
    }

}